src/commands/sourcemaps/renderer.ts

```diff
@@ -32,8 +34,45 @@ export const renderRetriedUpload = (payload: Payload, errorMessage: string, atte
   return chalk.yellow(`[attempt ${attempt}] Retrying sourcemap upload ${sourcemapPathBold}: ${errorMessage}\n`)
 }
 
-export const renderSuccessfulCommand = (fileCount: number, duration: number) =>
-  chalk.green(`${ICONS.SUCCESS} Uploaded ${fileCount} files in ${duration} seconds.\n`)
+export const renderSuccessfulCommand = (statuses: UploadStatus[], duration: number, dryRun: boolean) => {
+  const results = new Map<UploadStatus, number>()
+  statuses.forEach((status) => {
+    if (!results.has(status)) {
+      results.set(status, 0)
+    }
+    results.set(status, results.get(status)! + 1)
+  })
+
+  const output = []
+
+  if (results.get(UploadStatus.Success)) {
+    if (dryRun) {
+      output.push(
+        chalk.green(
+          `${ICONS.SUCCESS} [DRYRUN] Handled ${results.get(UploadStatus.Success)} (out of ${
+            statuses.length
+          }) sourcemaps with success in ${duration} seconds.`
+        )
+      )
+    } else {
+      output.push(
+        chalk.green(
+          `${ICONS.SUCCESS} Uploaded ${results.get(UploadStatus.Success)} (out of ${
+            statuses.length
+          }) sourcemaps in ${duration} seconds.`
+        )
+      )
+    }
+  }
+  if (results.get(UploadStatus.Failure)) {
+    output.push(chalk.red(`${ICONS.FAILED} ${results.get(UploadStatus.Failure)} files failed to upload.`))
+  }
+  if (results.get(UploadStatus.Skipped)) {
+    output.push(chalk.yellow(`${ICONS.WARNING}  ${results.get(UploadStatus.Skipped)} files were ignored.`))
+  }
+
+  return output.join('\n') + '\n'
+}
```

Comment:
**Super NIT**: you could pluralize outputs with an helper function to simplify the process, I think it would make the output more readable.

```js
const pluralize = (nb, singular, plural) => {
  if (nb >= 2) {
    return `${nb} ${plural};
  }
  return `${nb} ${singular}`;
} 
```

And then use it like:
```js
output.push(chalk.yellow(`${ICONS.WARNING}  ${pluralize(results.get(UploadStatus.Skipped), 'file was', 'files were')} ignored.`))
```

Reply:
ah nice! Will do, it will look much better, thanks!